### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by jcam3ron**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Cassandra Java Driver Kerberos Authenticator
+
```
Comment (kgreav) [inline README.md:2]: Obv need to add more details here... essentially what's in the javadoc should be a good start.

File: src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+ * for more details).
+ *
+ * The following example JAAS configuration demonstrates authentication via a keytab:
+ * <pre>{@code
+ * CassandraJavaClient {
+ *     com.sun.security.auth.module.Krb5LoginModule required
+ *          storeKey=true
+ *          principal="principal@MYREALM.COM"
+ *          useKeyTab=true
+ *          keyTab="/path/to/principal.keytab"
```
Comment (kgreav) [inline src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java]:
```suggestion
 *          keyTab="/path/to/principal.keytab";
```

Missing semi-colon leads to a really cryptic error:
```
Exception in thread "main" com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /10.150.0.1:9042 (com.datastax.driver.core.exceptions.ConnectionException: [/10.150.0.1:9042] Unexpected error during transport initialization (java.lang.SecurityException: java.io.IOException: Configuration Error:
	Line 7: expected [option key])))
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:268)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:107)
	at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1652)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1571)
	at com.datastax.driver.core.Cluster.init(Cluster.java:208)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
	at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
	at com.instaclustr.cassandra.driver.auth.TestClient.main(TestClient.java:42)
```

File: src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        private static final String JAAS_CONFIG_ITEM_NAME = "CassandraJavaClient";
+        private static final String[] SASL_MECHANISMS = new String[]{"GSSAPI"};
+
+        private final Subject subject;
+        private final SaslClient saslClient;
+
+        private KerberosAuthenticator(String authorizationId, String saslProtocol, InetSocketAddress host, Map<String, ?> saslProperties) {
+
+            try {
+                final LoginContext loginContext = new LoginContext(JAAS_CONFIG_ITEM_NAME);
```
Comment (kgreav) [inline src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java]: We should probably catch exceptions here for errors in the jaas config and print a nice error message. See my example above with the missing semi-colon. I think if we catch an IOException here it'll work...

Comment (kgreav) [reply src/main/java/com/instaclustr/cassandra/driver/auth/KerberosAuthProvider.java]: Also seems like any exceptions that get thrown here are really messy when presented to the client:
```
Exception in thread "main" com.datastax.driver.core.exceptions.NoHostAvailableException: All host(s) tried for query failed (tried: /10.150.0.1:9042 (com.datastax.driver.core.exceptions.ConnectionException: [/10.150.0.1:9042] Unexpected error during transport initialization (java.lang.RuntimeException: javax.security.sasl.SaslException: GSS initiate failed [Caused by GSSException: No valid credentials provided (Mechanism level: Failed to find any Kerberos tgt)])))
	at com.datastax.driver.core.ControlConnection.reconnectInternal(ControlConnection.java:268)
	at com.datastax.driver.core.ControlConnection.connect(ControlConnection.java:107)
	at com.datastax.driver.core.Cluster$Manager.negotiateProtocolVersionAndConnect(Cluster.java:1652)
	at com.datastax.driver.core.Cluster$Manager.init(Cluster.java:1571)
	at com.datastax.driver.core.Cluster.init(Cluster.java:208)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:376)
	at com.datastax.driver.core.Cluster.connectAsync(Cluster.java:355)
	at com.datastax.driver.core.Cluster.connect(Cluster.java:305)
	at com.instaclustr.cassandra.driver.auth.TestClient.main(TestClient.java:42)
```
this was another example where i don't have a tgt

**Review by jordanbraiuka — APPROVED** [review]
Looks good to me

**Pull request closed without merging**